src/datasource/components/ConfigEditor.test.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { ConfigEditor } from './ConfigEditor';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { AwsAuthType } from '@grafana/aws-sdk';
+import { DataSourceSettings } from '@grafana/data';
+import { TwinMakerDataSourceOptions, TwinMakerSecureJsonData } from 'datasource/types';
+
+const datasourceOptions: DataSourceSettings<TwinMakerDataSourceOptions, TwinMakerSecureJsonData> = {
+  id: 0,
+  uid: 'test',
+  orgId: 0,
+  name: 'twinmaker',
+  typeLogoUrl: '',
+  type: 'datasource',
+  typeName: 'Datasource',
+  access: 'server',
+  url: 'http://localhost',
+  user: '',
+  database: '',
+  basicAuth: false,
+  basicAuthUser: '',
+  isDefault: false,
+  jsonData: {},
+  readOnly: false,
+  withCredentials: false,
+  secureJsonFields: {},
+  password: 'test',
+  basicAuthPassword: 'password',
+};
+const workspacesMock = jest.fn(() => Promise.resolve([{ value: 'test1', label: 'test1' }]));
+
+jest.mock('common/datasourceSrv', () => ({
+  ...jest.requireActual('common/datasourceSrv'),
+  getTwinMakerDatasource: jest.fn(() => ({
+    info: {
+      listWorkspaces: workspacesMock,
+    },
+  })),
+}));
+
+function setup() {
```

Comment:
According to the style guide: 

> Directly calling methods from `userEvent` may not be supported in future versions. As such, it's necessary to first call `userEvent.setup()` prior to the tests. This method returns a `userEvent` instance, complete with all its methods. This setup process can be simplified using a utility function:

https://github.com/grafana/grafana/blob/main/contribute/style-guides/testing.md